templates/extras/simp.md

```diff
@@ -62,222 +100,301 @@ lemmas". Here are some examples of `simp` lemmas in mathlib:
-can be called at will with `simp with new_attr`. This can often replace lengthy 
-`simp only [...]` calls, and facilitate easier-to-read code. Some examples of common usage are 
+you can make your own `@[simp]`-like attribute, but with a key difference:
+lemmas tagged with `@[new_attr]` are _not_ in the default set of `simp` lemmas.
+Instead, they are included using the syntax `simp with new_attr`. This can often replace lengthy 
+`simp only [...]` calls and facilitate easier-to-read code. Some examples of common usage are 
 [`mfld_simps`](https://leanprover-community.github.io/mathlib_docs/find/simp_attr.mfld_simps/src),
 and [`field_simps`](https://leanprover-community.github.io/mathlib_docs/find/simp_attr.field_simps/src).
 
-`simp {single_pass := tt}` -- this `single_pass` is a config option,
-one of around 16 at the time of writing. One can use `single_pass` to
-avoid loops which would otherwise occur.
-
-Searching for `structure simp_config` in the file
-`init/meta/simp_tactic.lean` in core Lean reveals other config
-options, not all of them documented, and most of them not very
-relevant for the average user.
-
+### Configuration options
+
+Both `simp` and `dsimp` can take additional configuration options using record syntax.
+For example, `simp {single_pass := tt}` runs `simp` with the `single_pass` configuration option set to true.
+One can use `single_pass` to avoid loops which would otherwise occur.
+
+Searching for `structure dsimp_config` and `structure simp_config` in
+the core Lean file `init/meta/simp_tactic.lean` reveals other
+configuration options.  Most of them not very relevant for the average user,
+and some of them are not fully documented.  These are reproduced in the
+following table, where the default value for a configuration option
+for `simp` or `dsimp` is given in the respective column -- if no
+default value is present, that option is unavailable.
+The "max" default value refers to `simp.default_max_steps`, which is currently `10000000`.
+
+| Option | `simp` | `dsimp` | Description |
+| --- | --- | --- | --- |
+| `contextual` | `ff` | | Use additional `simp` lemmas based on the context of the current subexpression (see example below)  |
+| `single_pass` | `ff` | `ff` | Visit each subterm no more than once |
+| `md` | | `reducible` | Reduction mode: how aggressively constants are replaced with their definitions (`all`, `semireducible`, `instances`, `reducible`, or `none`) |
+| `max_steps` | max | max | The maximum number of steps allowed before failing |
+| `canonize_instances` | `tt` | `tt` | Replace each instance with a canonical defeq one |
+| `canonize_proofs` | `ff` |  |  |
+| `fail_if_unchanged` | `tt` | `tt` | Fail if no simplifications applied |
+| `eta` | `tt` | `tt` | Allow eta-equivalence: `(λ x, F x)` ↝ `F` |
+| `zeta`| `tt` | `tt` | Do zeta-reductions: `let x : a := b in c` ↝ `c[x/b]` |
+| `beta` | `tt` | `tt` | Do beta-reductions: `(λ x, E) y` ↝ `E[x/y]` |
```

Comment:
```suggestion
| `zeta`| `tt` | `tt` | Do zeta-reductions: `let x : a := b in c` ↝ `c[x := b]` |
| `beta` | `tt` | `tt` | Do beta-reductions: `(λ x, E) y` ↝ `E[x := y]` |
```
Substitution is usually/often written the other way around: `c[b/x]` (see e.g. the link below). The above suggestion is unambiguous notation for substitution (but maybe confusing since we also use it in `let`?
https://math.stackexchange.com/questions/273294/substitution-in-lambda-calculus

(also fix the occurrence below the table)

Reply:
Thanks, I'm happy to replace that with less ambiguous notation.

For what it's worth, it's lifted from https://github.com/leanprover-community/lean/blob/master/library/init/meta/simp_tactic.lean#L132